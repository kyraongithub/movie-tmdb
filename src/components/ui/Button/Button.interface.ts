import type { ButtonHTMLAttributes } from 'react';

export interface ButtonPropsInterface
  extends ButtonHTMLAttributes<HTMLButtonElement> {}
