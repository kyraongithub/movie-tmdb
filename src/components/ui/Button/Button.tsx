import type { ButtonPropsInterface } from './Button.interface';

const Button = (props: ButtonPropsInterface) => {
  return (
    <button className="btn cursor-pointer" {...props}>
      {props.children}
    </button>
  );
};

export default Button;
