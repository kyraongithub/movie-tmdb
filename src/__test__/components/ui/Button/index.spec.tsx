import { render } from '@testing-library/react';
import Button from '../../../../components/ui/Button';

describe('Button', () => {
  it('render button component', () => {
    const { container } = render(<Button>Button</Button>);
    expect(container).toMatchSnapshot();
  });
});
