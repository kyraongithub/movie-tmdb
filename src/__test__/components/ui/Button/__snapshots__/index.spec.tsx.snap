// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > render button component 1`] = `
<div>
  <button>
    Button
  </button>
</div>
`;
